src/programs/hadeswap/price.ts

```diff
@@ -1,9 +1,59 @@
-import { hadeswap } from 'hadeswap-sdk';
 import Big from 'big.js';
-const {
-  types: { OrderType, BondingCurveType },
-  helpers: { calculateNextSpotPrice },
-} = hadeswap;
+import { HadeswapBondingCurveType, HadeswapOrderType } from './constants';
+
+// NB: copied from hadeswap-sdk
+export const calculateNextSpotPrice = ({
+  orderType,
+  spotPrice,
+  delta,
+  bondingCurveType,
+  counter,
+}: {
+  orderType: HadeswapOrderType;
+  spotPrice: number;
+  delta: number;
+  bondingCurveType: HadeswapBondingCurveType;
+  counter: number;
+}): number => {
+  if (bondingCurveType === HadeswapBondingCurveType.Linear) {
+    let current_price = spotPrice; // 1
+
+    const targetCounter =
+      counter + (orderType === HadeswapOrderType.Buy ? 1 : -1);
+    if (targetCounter >= 0) {
+      // 0
+      for (let i = 0; i < Math.abs(targetCounter); i++) {
+        current_price += delta;
+      }
+    } else {
+      for (let i = 0; i < Math.abs(targetCounter); i++) {
+        current_price -= delta;
+      }
+    }
+    return current_price;
+  } else if (bondingCurveType === HadeswapBondingCurveType.Exponential) {
+    const newCounter =
+      orderType === HadeswapOrderType.Buy ? counter + 1 : counter - 1;
+    let newDelta =
+      newCounter > 0 ? (delta + 1e4) / 1e4 : 1 / ((delta + 1e4) / 1e4);
+
+    return spotPrice * Math.pow(newDelta, Math.abs(newCounter));
+  } else if (bondingCurveType === HadeswapBondingCurveType.XYK) {
+    // const deltaCorrected = delta - counter;
+
+    const nftTokensBalance = delta * spotPrice;
+    const counterUpdated =
+      orderType === HadeswapOrderType.Buy ? counter : counter - 1;
+    const currentDelta = delta + 1 - counterUpdated;
+    const diffAmount = (counterUpdated * nftTokensBalance) / currentDelta;
+    const newNftTokensBalance = nftTokensBalance + diffAmount;
+
+    return orderType === HadeswapOrderType.Buy
+      ? newNftTokensBalance / (currentDelta - 1)
+      : newNftTokensBalance / (currentDelta + 1);
+  }
+  return 0;
+};
```

Comment:
You pulled this from the bundled code here right? https://www.npmjs.com/package/hadeswap-sdk?activeTab=code

Reply:
from here, updated the comment
 https://github.com/hadeswap-solana/hadeswap-sdk-public/blob/66ab430a7a11bd014804d48a7f5d2e7fd906f9ac/src/hadeswap-core/helpers.ts#L86